addons/sale_loyalty/models/sale_order.py

```diff
@@ -494,6 +494,7 @@ def _get_real_points_for_coupon(self, coupon, post_confirm=False):
             points += self.coupon_point_ids.filtered(lambda p: p.coupon_id == coupon).points
         # Points already used by rewards
         points -= sum(self.order_line.filtered(lambda l: l.coupon_id == coupon).mapped('points_cost'))
+        points = coupon.currency_id.round(points) if coupon.currency_id else float_round(points, precision_digits=2)
```

Comment:
It makes no sense to round using the currency actually, not every program is about direct currency.
I would guess rounding at the second decimal is probably enough.

Reply:
If the program doesn't use direct currency, won't the `currency_id` here be `None` in that case? My concern with hardcoding it to 2 for everybody is the countries that use 3 and 4 digit precision currencies.

Reply:
`currency_id` is actually required.
It goes both way, what if you have a program that requires n digits of precision but your currency only gives n-1.
Although this is an edge case and maybe I'm overthinking. Using `currency_id` will work in most cases.

Reply:
Oh, `currency_id` is indeed required, I had misinterpreted something.

If you have a program with a higher precision that your currency, is it currently possible to specify that? You mean if people just enter the higher precision values when specifying the points amounts?

Reply:
Yep exactly, but honestly it shouldn't be an issue.. the fix is ok we just need to remove the check on currency_id as it's redundant

Reply:
Done and retested for my use case.